java/fury-core/src/main/java/io/fury/ThreadSafeFury.java

```diff
@@ -31,7 +31,8 @@ public interface ThreadSafeFury {
 
   /**
    * Provide a context to execution operations on {@link Fury} directly and return the executed
-   * result.
+   * result. Do NOT use this api to register class, for registering class here will have thread
+   * safety issue. The registration won't propagate to other threads.
```

Comment:
A better option would be implement `ThreadLocalFury(Function<FuryBuilder, Fury> furyFactory)` constructor, to avoid users to understand the complexities of `ClassLoader `. 

In first method, users must create Fury with classloader set as pased clasloader:
```java
new ThreadLocalFury(classloader -> {
  Fury f = Fury.builder().withClassLoader(classloader).build();
  f.register(XXX.class);
  return f;
})
```

It's very possible that users forget to invoke `withClassLoader` method, and introduce poential bug.

Reply:
For the implementation, we can construct a new `Function<FuryBuilder, Fury> furyFactory` by creating `FuryBuilder` with classloader set inside in advance.

Reply:
Got it.